src/renderer/contexts/ChatContext.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useCallback, useState } from 'react'
+
+import { getChat, markChatAsSeen, saveLastChatId } from '../backend/chat'
+
+import type { PropsWithChildren } from 'react'
+import type { T } from '@deltachat/jsonrpc-client'
+
+export enum ChatView {
+  Map,
+  Media,
+  MessageList,
+}
+
+export type SetView = (nextView: ChatView) => void
+
+export type SelectChat = (accountId: number, chatId: number) => Promise<void>
+
+export type UnselectChat = () => void
+
+export type ChatValue = {
+  accountId?: number
+  activeView: ChatView
+  chat?: T.FullChat
+  chatId?: number
+  selectChat: SelectChat
+  setChatView: SetView
+  unselectChat: UnselectChat
+}
+
+export const ChatContext = React.createContext<ChatValue | null>(null)
+
+export const ChatProvider = ({ children }: PropsWithChildren<{}>) => {
+  // @TODO: Explore how to refactor account state into its own context
```

Comment:
maybe a a context for the screen controller, but might also make sense to  duplicate it as measure to prevent it going out of sync?
or have some use(Layout)Effect hook that removes the chat data as soon as the accountId changes?

Reply:
There's different approaches we could take:

1. Call the `unselectChat` method on the `ChatContext` whenever `unselectAccount` in `ScreenController` gets called
2. Use `key={this.selectedAccountId}` to force `ChatContext` to rebuild with default values
3. Use an React effect to internally call `unselectChat` in `ChatContext` whenever account id changes

Currently I'm using the 2nd option, even though it is a little bit implicit. Probably doing 1. or 3. is better.

Reply:
Looking at it again I think it's fine as it is right now (using `key`), later we can make it more explicit, for example with 1. but I think we might need to touch too much of the account logic for this then and this is out of scope of this PR

Reply:
In any case, I've removed `accountId` from the `ChatProvider`, now it is passed in as an prop which makes more sense

Reply:
I think 2 (react `key` "hack") makes most sense